pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -53,7 +54,7 @@ private Metrics() { // Cannot be instantiated
      *
      * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed
      */
-    public static double get(ClassMetricKey key, ASTAnyTypeDeclaration node) {
+    public static double get(MetricKey<ClassMetric> key, ASTAnyTypeDeclaration node) {
```

Comment:
I don't really get your point. `Metrics` is really just a user interface, why would it need to be something other than a singleton? Btw testing for metrics is quite straightforward, see #482